src/base/BaseStrategy.sol

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IERC20, Math, SafeERC20} from "src/Common.sol";
+import {Vault} from "src/Vault.sol";
+
+/**
+ * @title BaseStrategy
+ * @author Yieldnest
+ * @notice This contract is a base strategy for any underlying protocol.
+ * vault.
+ */
+abstract contract BaseStrategy is Vault {
```

Comment:
this should probably not inherit Vault but BaseVault

Reply:
@danoctavian i did this so fees are also included here, thats the main difference between BaseVault and Vault right?

Reply:
i'd keep that out not all strategies have fees